k2/python/csrc/torch/doc/any.h

```diff
@@ -0,0 +1,416 @@
+)doc";
+
+static constexpr const char *kRaggedAnyRequiresGradPropDoc = R"doc(
+Return ``True`` if gradients need to be computed for this tensor.
+Return ``False`` otherwise.
+
+>>> import torch
+>>> import k2.ragged as k2r
+>>> a = k2r.Tensor([[1]], dtype=torch.float32)
+>>> a.requires_grad
+False
+>>> a.requires_grad = True
+>>> a.requires_grad
+True
+)doc";
+
+static constexpr const char *kRaggedAnyGradPropDoc = R"doc(
+This attribute is ``None`` by default. PyTorch will set it
+during ``backward()``.
+
+The attribute will contain the gradients computed and future
+calls to ``backward()`` will accumulate (add) gradients into it.
+
+>>> import k2.ragged as k2r
+>>> a = k2r.Tensor([[1, 2], [3], [5, 6], []], dtype=torch.float32)
+>>> a.requires_grad_(True)
+[ [ 1 2 ] [ 3 ] [ 5 6 ] [ ] ]
+>>> b = a.sum()
+>>> b
+tensor([ 3.,  3., 11.,  0.], grad_fn=<SumFunction>>)
+>>> c = b * torch.arange(4)
+>>> c.sum().backward()
+>>> a.grad
+tensor([0., 0., 1., 2., 2.])
+)doc";
+
+static constexpr const char *kRaggedAnyRequiresGradMethodDoc = R"doc(
+Change if autograd should record operations on this tensor: Set
+this tensor's :attr:`requires_grad` attribute **in-place**.
+
+Note:
+  If this tensor is not a float tensor, PyTorch will throw a
+  RuntimeError exception.
+
+Caution:
+  This method ends with an underscore, meaning it changes this tensor
+  **in-place**.
+
+>>> import torch
+>>> import k2.ragged as k2r
+>>> a = k2r.Tensor([[1]], dtype=torch.float64)
+>>> a.requires_grad
+False
+>>> a.requires_grad_(True)
+[ [ 1 ] ]
+>>> a.requires_grad
+True
+
+Args:
+  requires_grad:
+    If autograd should record operations on this tensor.
+Returns:
+  Return this tensor.
+)doc";
+
+static constexpr const char *kRaggedAnySumDoc = R"doc(
+Compute the sum of sublists over the last axis of this tensor.
+
+Note:
+  If a sublist is empty, the sum for it is the provided
+  ``initial_value``.
+
+Note:
+  This operation supports autograd if this tensor is a float tensor,
+  i.e., with dtype being torch.float32 or torch.float64.
+
+>>> import k2.ragged as k2r
+>>> a = k2r.Tensor('[ [[1 2] [] [5]]  [[10]] ]', dtype=torch.float32)
+>>> a.requires_grad_(True)
+[ [ [ 1 2 ] [ ] [ 5 ] ] [ [ 10 ] ] ]
+>>> b = a.sum()
+>>> c = (b * torch.arange(4)).sum()
+>>> c.backward()
+>>> a.grad
+tensor([0., 0., 2., 3.])
+>>> b
+tensor([ 3.,  0.,  5., 10.], grad_fn=<SumFunction>>)
+>>> c
+tensor(40., grad_fn=<SumBackward0>)
+
+Args:
+  initial_value:
+    This value is added to the sum of each sublist. So when
+    a sublist is empty, its sum is this value.
+Returns:
+  Return a 1-D tensor with the same dtype of this tensor
+  containing the computed sum.
+)doc";
+
+static constexpr const char *kRaggedAnyNumelDoc = R"doc(
+Returns:
+  Return number of elements in this tensor. It equals to
+  `self.data.numel()`.
+>>> import torch
+>>> import k2.ragged as k2r
+>>> a = k2r.Tensor([[1], [], [3, 4, 5, 6]])
+>>> a.numel()
+5
+>>> b = k2r.Tensor('[ [[1] [] []]  [[2 3]]]')
```

Comment:
It is better to use the same constructor in one example.

Reply:
Thanks, fixed.